src/Commands/DrupalCommands.php

```diff
@@ -270,20 +272,37 @@ public function drushSetup(array $options = [
      * >       sync: '../config/sync'
      * >       prod: '../config/prod'
      *
+     * The settings override file name can be changed in the Task Runner
+     * configuration by setting the "drupal.site.settings_override_file" property.
+     *
      * @command drupal:settings-setup
      *
-     * @option root Drupal root.
+     * @option root                     Drupal root.
+     * @option sites-subdir             Drupal site subdirectory.
+     * @option settings-override-file   Drupal site settings override filename.
      *
      * @param array $options
      *
      * @return \Robo\Collection\CollectionBuilder
      */
     public function settingsSetup(array $options = [
-      'root' => InputOption::VALUE_REQUIRED,
+        'root' => InputOption::VALUE_REQUIRED,
+        'sites-subdir' => InputOption::VALUE_REQUIRED,
+        'settings-override-file' => InputOption::VALUE_REQUIRED,
     ])
     {
+        $settings_default_path = $options['root'] . '/sites/' . $options['sites-subdir'] . '/default.settings.php';
+        $settings_path = $options['root'] . '/sites/' . $options['sites-subdir'] . '/settings.php';
+        $settings_override_path = $options['root'] . '/sites/' . $options['sites-subdir'] . '/' . $options['settings-override-file'];
+
         return $this->collectionBuilder()->addTaskList([
-            $this->taskAppendConfiguration($options['root'].'/sites/default/default.settings.php', $this->getConfig())->setConfigKey('drupal.settings'),
+            $this->taskFilesystemStack()->copy($settings_default_path, $settings_path, true),
```

Comment:
Let's make the `force` configurable (last parameter) by passing a parameter to the command (so no configuration value in this case), and let's default it to false, since if we force it here we wipe out the whole settings.php file which will force a full site reinstall in order to have it regenerated by Drupal.